observable_operator.go

```diff
@@ -531,14 +533,21 @@ func (o *ObservableImpl) BufferWithTimeOrCount(timespan Duration, count int, opt
 
 		checkBuffer := func() {
 			mutex.Lock()
-			if len(buffer) != 0 {
-				if !Of(buffer).SendContext(ctx, next) {
-					mutex.Unlock()
-					return
+			defer mutex.Unlock()
+
+			length := len(buffer)
+			if length != 0 {
+				for i := 0; i < length; i += count {
```

Comment:
Mmh... 🤔 I agree that you have spotted a problem, thanks for that but I'm wondering whether it's exactly what we want.
Let's say if the length of `buffer` is 6 and count is 5, if I understand your implementation we will send two slices in a row:
* From elements to 0 to 5
* Element 6

Should we really send element 6 as part of the same `checkBuffer` execution or rather keep this element for the next execution of `checkBuffer`?

Perhaps we should use your idea but simply return after having sent a message and resliced `buffer`?

```go
checkBuffer := func() {
	mutex.Lock()
	defer mutex.Unlock()

	length := len(buffer)
	if length != 0 {
		if length > count {
			length = count
		}
		Of(buffer[:length]).SendContext(ctx, next)
		buffer = buffer[length:]
	}
}
```

What do you think?

Reply:
iiuc, the other language implementations cap the items at size count; I think this is what's being reported.  I'll double check with the other implementations.

Reply:
Ok, I checked w/ the other implementations and in reality, "count" is "max count".  The illustration for this project is a bit different than what's for Java

https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/buffer6.s.v3.png

The latter better reflects the proper semantics.

Reply:
> the other language implementations cap the items at size count; I think this is what's being reported

Not sure to understand, I wasn't really discussing this point.

In the example I described above, your implementation of `checkBuffer` will publish two items in a row:
* From elements to 0 to 5
* Element 6

My point is that I don't think we should publish element 6. Element 6 should be published during the next execution of `checkBuffer`

Reply:
Ok, understood.  I think the buffer should be flushed when stopped, otherwise the behavior should be as you say, leave element 6 in the buffer.  I'll update the PR.